client/js/api-reference/messages.mdx

```diff
@@ -1,45 +1,288 @@
+    return {"url": room_url, "token": token}
+```
+
+```python bot
+import argparse
+import asyncio
+
+def extract_arguments():
+    parser = argparse.ArgumentParser(description="Example")
+    parser.add_argument(
+        "-u", "--room-url", type=str, default=os.getenv("DAILY_SAMPLE_ROOM_URL", "")
+    )
+    parser.add_argument(
+        "-t", "--token", type=str, default=os.getenv("DAILY_SAMPLE_ROOM_TOKEN", None)
+    )
+    parser.add_argument(
+        "-p", "--prompt", type=str, default="You are a pirate captain"
+	)
+	parser.add_argument("-l", "--language", type=str, default="en-US")
+    return parser.parse_args()
+
+async def main():
+    args = extract_arguments()
+    print(f"room_url: {args.room_url}")
+
+    daily_transport = DailyTransport(
+        args.room_url,
+        args.token,
+        "Chatbot",
+        DailyParams(
+            audio_in_enabled=True,
+            audio_out_enabled=True,
+            vad_analyzer=SileroVADAnalyzer(),
+        ),
+    )
 
-Developers can build their own messages and create handlers to respond to messages received from a bot either by extending `RTVIClient` with their own class or by building a [helper](./helpers/introduction).
+    llm = GeminiMultimodalLiveLLMService(
+        api_key=os.getenv("GOOGLE_API_KEY"),
+        voice_id="Puck",
+        transcribe_user_audio=True,
+        system_instruction=SYSTEM_INSTRUCTION,
+		params=InputParams(
+			language=args.language,  # Pass preferred language to LLM
+		),
+    )
 
-### sendMessage(message: RTVIMessage)
+	messages = [ { role: "system", content: args.prompt } ]
+    context = OpenAILLMContext(messages)
+    context_aggregator = llm.create_context_aggregator(context)
 
-This is a "fire and forget" function to send an arbitrary message to a bot. It does not wait for a response, so nothing is returned.
+    # RTVI events for Pipecat client UI
+    rtvi = RTVIProcessor(config=RTVIConfig(config=[]), transport=daily_transport)
 
-Most clients will not use this function directly. They will use [actions](./actions) and [helper methods](./helpers), instead.
+    pipeline = Pipeline(
+        [
+            daily_transport.input(),
+            context_aggregator.user(),
+            rtvi,
+            llm,
+            daily_transport.output(),
+            context_aggregator.assistant(),
+        ]
+    )
 
-It is up to the transport layer to pack and unpack messages, send messages over the wire, and deliver messages to the SDK so that events can be emitted.
+    task = PipelineTask(
+        pipeline,
+        params=PipelineParams(allow_interruptions=True),
+        observers=[RTVIObserver(rtvi)],
+    )
 
-### Anatomy of a message
+    @rtvi.event_handler("on_client_ready")
+    async def on_client_ready(rtvi):
+        await rtvi.set_bot_ready()
+        # Kick off the conversation
+        await task.queue_frames([context_aggregator.user().get_context_frame()])
+
+    @daily_transport.event_handler("on_participant_left")
```

Comment:
stepping back: would it be better if this were an rtvi event?

Reply:
maybe. other than supporting the new behaviors and deprecating the ones going away, though, i didn't do a ton of work on the pipecat side of all this. maybe we do a better sweep of things like this in post?